### PR TITLE
Prevent scheduled event processor from being executed on forks

### LINKED ISSUE #36257: [QUERY] Scheduled Event Processor creates noise on forks
### Library name and version

All

### Query/Question

The Scheduled Event Processor github action is triggered also on forks according to the schedule and creates lots of failed actions noise on those. Would it be possible to conditionally skip those when the github token is not available? (which is only known on the upstream repo here) 

### Environment

_No response_

### PR DESCRIPTION
**Pull request by danielmarbach**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/36257

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: .github/workflows/scheduled-event-processor.yml [hunk 1]
```diff
@@ -20,6 +20,8 @@ permissions: {}
 
 jobs:
   event-handler:
+    # prevents the job from being executed on forks
+    if: github.repository == 'azure/azure-sdk-for-net'
```
Comment (jsquire) [inline .github/workflows/scheduled-event-processor.yml:24]: Unfortunately, this won't work as-is.   The file is pushed out from the tools repository into each of the other SDK repositories and the content can't change.

I wonder, though if we could do some kind of `startswith` for `azure/`?   @JimSuplizio, thoughts?

Comment (danielmarbach) [inline .github/workflows/scheduled-event-processor.yml:24]:
```suggestion
    if: startsWith(github.repository, 'azure') != true
```

Comment (danielmarbach) [reply .github/workflows/scheduled-event-processor.yml:24]: I think the job level secret check wouldn't work due to [this restriction,](https://github.com/orgs/community/discussions/26726#discussioncomment-3253118) but if you like, I can flip this PR to include the secrets check on every step

**Comment by danielmarbach** [comment]
@JimSuplizio well the thing is if it runs but gets skipped then there is no issue anymore from a fork perspective in respect to the noise failed runs create. Successful runs with skipped actions do not create notifications.

Just as an example

image

These are all the notifications I got until I learned that I have to manually disable this workflow.

I agree, this solution doesn't prevent it from running but it prevents everyone that forks having to disable this workflow or suppress all the notifications.

**Comment by danielmarbach** [comment]
Don't worry I appreciate you taking the time to reply. I didn't invest a lot of time into this anyway. My idea was mostly to just follow through with a potentially actionable idea as a PR suggestion.

I'm totally fine with things getting closed.

**Pull request closed without merging**